### PR TITLE
Lambda reserved concurrency

### PR DESCRIPTION
**Pull request by gmcrocetti**
Related with #3067. Still **missing** implementation for provisioned concurrency.

### CODE REVIEW

File: moto/awslambda/responses.py [hunk 1]
```diff
@@ -359,3 +372,31 @@ def _put_code(self):
             return 200, {}, json.dumps(resp)
         else:
             return 404, {}, "{}"
+
+    def _get_function_concurrency(self, request):
+        function_name = self.path.rsplit("/", 2)[-2]
+        resp = self.lambda_backend.get_function_concurrency(function_name)
+
+        if resp:
```
Comment (bblommers) [inline moto/awslambda/responses.py:380]: The if-else statements (for all three, get/put/delete) don't seem to be necessary - the backend always returns one value

Comment (gmcrocetti) [reply moto/awslambda/responses.py:380]: Ooops. I've forgot to validate function's name in the backend. We must return a 404 in this case, right ?

**Comment by gmcrocetti** [comment]
Hey @bblommers, tests are failing at "server mode". I have no idea why, any tips ?

File: moto/awslambda/responses.py (hunk 1 above)
Comment (bblommers) [reply moto/awslambda/responses.py]: Yes, correct. This is the error as returned from AWS:
```
botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the PutFunctionConcurrency operation: Function not found: [arn]
```

File: moto/awslambda/responses.py [hunk 2]
```diff
@@ -359,3 +372,38 @@ def _put_code(self):
...
+
+    def _get_function_concurrency(self, request):
+        path_function_name = self.path.rsplit("/", 2)[-2]
+        function_name = self.lambda_backend.get_function(path_function_name)
+
+        if function_name is None:
+            return 404, {}, "{}"
+
+        resp = self.lambda_backend.get_function_concurrency(path_function_name)
+        return 200, {}, json.dumps({"ReservedConcurrentExecutions": resp})
```
Comment (bblommers) [inline moto/awslambda/responses.py:384]: The concurrency-attribute might not be set yet, in which case the response would simply be `json.dumps({})`
Would be good to have a test case for this scenario as well

**Comment by bblommers** [comment]
@gmcrocetti The master branch is failing as well - don't think it's related to these changes :+1:

File: moto/awslambda/responses.py (hunk 2 above)
Comment (gmcrocetti) [reply moto/awslambda/responses.py:384]: Nice catch !!

Comment (gmcrocetti) [reply moto/awslambda/responses.py:384]: Hey @bblommers . Ran the same request using awscli in debug mode:
image
Looks like it's right :)

Comment (bblommers) [reply moto/awslambda/responses.py:384]: Fair enough. I ran it using boto3 directly, and then the field isn't returned at all - I guess that's the internal logic in boto that strips None-fields :)

**Comment by gmcrocetti** [comment]
Hi @bblommers , any updates ? Can we do smth to help you ?

**Comment by bblommers** [comment]
Hi @gmcrocetti, the main branch is now passing again. Would you mind merging in the latest changes? That should make this build pass as well

**Comment by gmcrocetti** [comment]
> Hi @gmcrocetti, the main branch is now passing again. Would you mind merging in the latest changes? That should make this build pass as well

Not working yet, failing due to a timeout. Can you please restart the job ?

**Review by bblommers — APPROVED** [review]

**Comment by bblommers** [comment]
All done @gmcrocetti - thanks for your PR!

**Comment by bblommers** [comment]
This is now part of moto >= 1.3.15.dev1029

**Pull request merged**